src/OmniSharp.Roslyn.CSharp/Services/Refactoring/V2/BaseCodeActionService.cs

```diff
@@ -98,41 +104,28 @@ private TextSpan GetTextSpan(ICodeActionRequest request, SourceText sourceText)
 
         private async Task CollectCodeFixesActions(Document document, TextSpan span, List<CodeAction> codeActions)
         {
-            Dictionary<TextSpan, List<Diagnostic>> aggregatedDiagnostics = null;
+            var analyzers = await this.analyzers.GetCurrentDiagnosticResult(new [] { document.Project.Id });
```

Comment:
this is a bit strange. it has snapshot. (Document) but doesn't get diagnostics from the snapshot but use some random one by giving (document.project.id ?)

I think it should give in the document.

also, variable name "analyzers" seems a bit confusing since it looks like actually resultPerAnalyzers

Reply:
I dont know how light bulb is invoked. but this seems using staled diagnostic data for code fixer. this has a quite big potential to throw since code fixer implicitly expects diagnostics given for the fixer is for the solution (document) given to the fixer. 

in this case, it might give diagnostics that is no longer exist for the document given to the fixer. and fixer might do wrong thing that can throw such as invalid cast and etc.